packages/TorneloScoresheet/__tests__/chessEngine.test.ts

```diff
@@ -949,3 +949,67 @@ describe('makeMove', () => {
     });
   });
 });
+
+// ----- chessEngine.isPawnPromotion() ----
+
+describe('isPromotion', function () {
+  const testCases = [
+    {
+      name: 'legal move non-promotion',
+      fen: 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1',
+      san: 'e4',
+      move: { from: 'e2', to: 'e4' },
+      promotion: false,
+    },
+    {
+      name: 'illegal move + promotion',
+      fen: 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1',
+      san: 'e8',
+      move: { from: 'e2', to: 'e8' },
+      promotion: true,
+    },
+    {
+      name: 'no piece on from',
+      fen: 'rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1',
+      san: 'e4',
+      move: { from: 'e3', to: 'e4' },
+      promotion: false,
+    },
+    // {
+    //   name: 'illegal promotion due to discovery',
+    //   fen: '1K6/2P2k2/8/8/5b2/8/8/8 w - - 0 1',
+    //   san: 'c8',
+    //   move: { from: 'c7', to: 'c8' },
+    //   promotion: false,
+    // },
+    {
+      name: 'illegal move non-capturing diagonal + promotion',
+      fen: '8/2P2k2/8/8/8/5K2/8/8 w - - 0 1',
+      san: 'b8',
+      move: { from: 'c7', to: 'b8' },
+      promotion: true,
+    },
+    {
+      name: 'legal promotion',
+      fen: '8/2P2k2/8/8/8/5K2/8/8 w - - 0 1',
+      san: 'c8',
+      move: { from: 'c7', to: 'c8' },
+      promotion: true,
+    },
+    {
+      name: 'legal capturing promotion',
+      fen: '1b6/2P2k2/8/8/5K2/8/8/8 w - - 0 1',
+      san: 'cxb8',
+      move: { from: 'c7', to: 'b8' },
+      promotion: true,
+    },
+  ];
+
+  testCases.forEach(function (testCase) {
+    it(testCase.name, function () {
+      expect(
+        chessEngine.isPawnPromotion(testCase.fen, testCase.move as PlySquares),
```

Comment:
Should this be `as MoveSquares`?

Reply:
Yeh weird as the test was still running and passing? Not sure how that's possible